arviz/tests/base_tests/test_stats_ecdf_utils.py

```diff
@@ -0,0 +1,159 @@
+import pytest
+
+import numpy as np
+import scipy.stats
+from arviz.stats.ecdf_utils import (
+    compute_ecdf,
+    ecdf_confidence_band,
+    _get_ecdf_points,
+    _simulate_ecdf,
+    _get_pointwise_confidence_band,
+)
+
+
+def test_compute_ecdf():
+    """Test compute_ecdf function."""
+    sample = np.array([1, 2, 3, 3, 4, 5])
+    eval_points = np.arange(0, 7, 0.1)
+    ecdf_expected = (sample[:, None] <= eval_points).mean(axis=0)
+    assert np.allclose(compute_ecdf(sample, eval_points), ecdf_expected)
+    assert np.allclose(compute_ecdf(sample / 2 + 10, eval_points / 2 + 10), ecdf_expected)
+
+
+@pytest.mark.parametrize("difference", [True, False])
+def test_get_ecdf_points(difference):
+    """Test _get_ecdf_points."""
+    # if first point already outside support, no need to insert it
+    sample = np.array([1, 2, 3, 3, 4, 5, 5])
+    eval_points = np.arange(-1, 7, 0.1)
+    x, y = _get_ecdf_points(sample, eval_points, difference)
+    assert np.array_equal(x, eval_points)
+    assert np.array_equal(y, compute_ecdf(sample, eval_points))
+
+    # if first point is inside support, insert it if not in difference mode
+    eval_points = np.arange(1, 6, 0.1)
+    x, y = _get_ecdf_points(sample, eval_points, difference)
+    assert len(x) == len(eval_points) + 1 - difference
+    assert len(y) == len(eval_points) + 1 - difference
+
+    # if not in difference mode, first point should be (eval_points[0], 0)
+    if not difference:
+        assert x[0] == eval_points[0]
+        assert y[0] == 0
+        assert np.allclose(x[1:], eval_points)
+        assert np.allclose(y[1:], compute_ecdf(sample, eval_points))
+        assert x[-1] == eval_points[-1]
+        assert y[-1] == 1
+
+
+@pytest.mark.parametrize(
+    "dist", [scipy.stats.norm(3, 10), scipy.stats.binom(10, 0.5)], ids=["continuous", "discrete"]
+)
+@pytest.mark.parametrize("seed", [32, None])
+def test_simulate_ecdf(dist, seed):
+    """Test _simulate_ecdf."""
+    ndraws = 1000
+    eval_points = np.arange(0, 1, 0.1)
+
+    rvs = dist.rvs
+
+    if seed is not None:
+        random_state = np.random.RandomState(seed)
+        ecdf = _simulate_ecdf(ndraws, eval_points, rvs, random_state=random_state)
+        random_state = np.random.RandomState(seed)
+        ecdf_expected = compute_ecdf(np.sort(rvs(ndraws, random_state=random_state)), eval_points)
+    else:
+        np.random.seed(87)
+        ecdf = _simulate_ecdf(ndraws, eval_points, rvs)
+        np.random.seed(87)
+        ecdf_expected = compute_ecdf(np.sort(rvs(ndraws)), eval_points)
```

Comment:
I wouldn't test that, scipy will eventually stop using the global state (see https://github.com/scipy/scipy/issues/14322 for example) in which case this would start breaking for no reason.